src/directives/input-restriction/input-restriction.ts

```diff
@@ -0,0 +1,94 @@
+import { Directive, ElementRef, HostListener, Input } from '@angular/core';
+@Directive({
+  selector: '[appInputRestriction]'
+})
+export class InputRestrictionDirective {
+  inputElement: ElementRef;
+
+  @Input('appInputRestriction') appInputRestriction: string;
+  arabicRegex = '[\u0600-\u06FF]';
+
+  constructor(el: ElementRef) {
+    this.inputElement = el;
+  }
+
+  @HostListener('keypress', ['$event'])
+  onKeyPress(event) {
+    if (this.appInputRestriction === 'integer') {
+      this.integerOnly(event);
+    } else if (this.appInputRestriction === 'noSpecialChars') {
+      this.noSpecialChars(event);
+    }
+  }
+
+  integerOnly(event) {
+    const e = <KeyboardEvent>event;
+    if (e.key === 'Tab' || e.key === 'TAB') {
+      return;
+    }
+    if (
+      [46, 8, 9, 27, 13, 110].indexOf(e.keyCode) !== -1 ||
```

Comment:
Could we pull these magic numbers out into an enum representing what keys they represent?

```typescript
enum KeyCode {
  A = 65,
  C = 67
}
```
and so forth?